Standards/scs-0217-v1-baseline-cluster-security.md

```diff
@@ -0,0 +1,145 @@
+type: Standard
+status: Draft
+track: KaaS
+---
+
+## Introduction
+
+A Kubernetes instance is provided as a cluster, which consists of a set of machines,
+so-called nodes. These clusters are configured depending on the needs of its users, which
+is possible due to the flexibility of Kubernetes.
+Due to the regular changes, there are always new security features to deploy and use in Kubernetes.
+Nevertheless, a provider (or even a customer) needs to take action in order to achieve a
+hardened, secure cluster due to the myriad of configurations possible. This is especially
+the case since Kubernetes ships with insecure features and configurations out of the box,
+which will need to be mitigated by an administrator with the proper knowledge.
+Hardened, secure Kubernetes clusters are desirable regardless of the possible threat model,
+since higher security doesn't necessarily mean higher complexity in this case.
+
+## Terminology
+
+| Term | Meaning                     |
+|------|-----------------------------|
+| TLS  | Transport Layer Security    |
+| CA   | Certificate Authority       |
+| CSR  | Certificate Signing Request |
+
+## Motivation
+
+Kubernetes clusters are highly configurable, which also gives rise to different security
+problems, if the configuration isn't done properly.
+These security risks can potentially be exposed in many different parts of a cluster, e.g.
+different APIs, authorization and authentication procedures or even Pod privilege mechanisms.
+In order to mitigate these problems, different steps and mechanisms could be used to increase
+the security of a Kubernetes setup.
+
+## Design Considerations
+
+### External CA
+
+Kubernetes provides an API to provision TLS certificates that can be signed by a CA.
+This CA can be controlled by the cluster provider, which enables much more tight control
+over the clusters communication and therefore also better controllable security.
+
+In order to do this, the CA certificate bundle needs to be added to the trusted certificates
+of the server.
+To provide a certificate, the following steps need to be undertaken:
+
+1. Create a CSR
+2. Send the CSR manifest to the k8s API
+3. Approve the CSR
+4. Sign CSR with your CA
+5. Upload the signed certificate to the server
+
+This certificate could now be used by a user in a pod in order to provide a trusted certificate.
+
+It is also possible for the Kubernetes controller manager to provide the signing functionality.
+To enable this, `--cluster-signing-cert-file` and `--cluster-signing-key-file` need to be set with
+a reference to the CA keypair, which was used in the previous example to sign a CSR.
+
+### Protected Kubernetes endpoints
+
+In order to secure a Kubernetes cluster, the protection of endpoints is important.
+To do this, different approaches can be taken.
+
+#### TLS for all internal/API traffic
+
+It is already expected by Kubernetes that all API communication internally is encrypted with TLS.
```

Comment:
where does this expectation come from? it's not mentioned as a requirement in this standard prior to this sentence.

I think we should add this.

Reply:
https://kubernetes.io/docs/tasks/administer-cluster/securing-a-cluster/#use-transport-layer-security-tls-for-all-api-traffic
From here, but I can mention this if that would make it more clear.